CHANGELOG.rst

```diff
@@ -1,3 +1,27 @@
+- Fish now asks the terminal to speak the keyboard input protocol CSI u in combination with XTerm's ``modifyOtherKeys`` and Kitty's progressive enhancements.
+  This allows to bind almost any key, including arbitrary combinations of modifiers ctrl, alt and shift.
+- There is a new syntax for specifying keys for builtin ``bind``.
+  The new syntax introduces modifier names and names for some keys that don't have an obvious and printable Unicode code point. The old syntax remains mostly supported but the new one is preferred.
+  - Existing bindings that use the new names have a different meaning now.
+    For example
+    - ``bind up 'do something'`` maps the up arrow instead of the two keys ``u`` and ``p``.
+    - ``bind ctrl-x,alt-c 'do something'`` maps a sequence of two key chords.
+  - To minimize breakage, ``KEYS`` argument to ``bind`` is parsed in the old syntax in two cases:
+    - If ``KEYS`` starts with a raw escape character (`\e`) or a raw ASCII control character (`\c`).
+    - If it contains none of ``,`` or ``-`` and it is not a named key.
+  Since ``,`` and ``-`` act as separators, they need to be given as ``comma`` and ``minus`` in ambiguous cases.
+- Specifying key names as terminfo name (``bind -k``) is deprecated and may be removed in a future version.
+- Legacy bindings that use raw escape sequences instead of named modifiers/keys continue to be included in the default configuration.
+  If your terminal supports CSI u, you can remove legacy bindings (including terminfo bindings) by setting the `fish_legacy_bindings` variable to ``false`` (on shell startup or before switching key bindings).
```

Comment:
I'm not sure why this mechanism is there already.

I don't see how keeping the old-school bindings along hurts, so I would leave these unconditional for now.

Or is there anything I'm missing?

(additionally: we typically use "0" and "1" for bools)

Reply:
`fish_legacy_bindings=false` is mainly to test that all the keys work as expected, I'm fine with removing it.
The only drawback of the old-school bindings is that they pollute the `bind` output (which is not much but also not nothing).